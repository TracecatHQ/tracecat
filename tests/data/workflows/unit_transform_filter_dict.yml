title: Filter data
description: Test to filter data
entrypoint:
  ref: filter_empty
inputs: #2D list
  matrix:
    - signal: a
      data: [1, 2, 3]
    - signal: b
      data: [4, 5, 6]
    - signal: c
      data: []
    - signal: d
      data: [7, 8, 9]
    - signal: e
      data: [10, 11, 12]
    - signal: f
      data: []

actions:
  - ref: filter_empty
    action: core.transform.filter
    args:
      # List of objects to filter
      items: ${{ INPUTS.matrix }}
      python_lambda: "lambda row: len(row['data']) > 0"
