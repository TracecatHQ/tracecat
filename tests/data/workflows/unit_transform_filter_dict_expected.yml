ACTIONS:
  filter_empty:
    result:
      - signal: a
        data: [1, 2, 3]
      - signal: b
        data: [4, 5, 6]
      - signal: d
        data: [7, 8, 9]
      - signal: e
        data: [10, 11, 12]
    result_typename: list

INPUTS:
  matrix:
    - signal: a
      data: [1, 2, 3]
    - signal: b
      data: [4, 5, 6]
    - signal: c
      data: []
    - signal: d
      data: [7, 8, 9]
    - signal: e
      data: [10, 11, 12]
    - signal: f
      data: []
TRIGGER:
